### PR TITLE
Update documentations and setups to Python 3.9

### LINKED ISSUE #14863: Python is not completely bumped to 3.9 in CI
I landed https://github.com/openxla/iree/commit/7d7391f1f989d77d1b4c6b02242f51efee93c084 to unblock integration tests and benchmarks; I have https://github.com/openxla/iree/pull/14860 as a follow-up to bump all related configs and docs about Python to 3.9, but there are errors in CI: https://github.com/openxla/iree/actions/runs/6006208662/job/16290354051?pr=14860

It looks like some env should be updated as well but I don't know what to do.

### PR DESCRIPTION
**Pull request by hanhanW**
The python minimum support was bumped to 3.9 in https://github.com/openxla/iree/pull/14810 This is a follow-up to update other documents and setups to Python 3.9.

Fixes https://github.com/openxla/iree/issues/14863

### CODE REVIEW

File: .github/workflows/validate_and_publish_release.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
         id: set_up_python
         uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435  # v4.5.0
         with:
-          python-version: "3.8"
+          python-version: "3.9"
```
Comment (ScottTodd) [inline .github/workflows/validate_and_publish_release.yml:40]: Here are a few more spots to remove "3.8":

https://github.com/openxla/iree/blob/88c94c19fb57205f98980a91f32b23045159c9c9/.github/workflows/build_package.yml#L3-L8

https://github.com/openxla/iree/blob/88c94c19fb57205f98980a91f32b23045159c9c9/build_tools/python_deploy/build_linux_packages.sh#L68

**Comment by hanhanW** [comment]
I keep getting failures on `pytype` check; I can't reproduce it locally. Can I get some eyes from someone or someone can help fix it?

```
Checking .py files in integrations/.*
ninja: Entering directory `.pytype'
ninja: error: build.ninja:13: multiple rules generate /home/runner/work/iree/iree/.pytype/pyi/setup.pyi

Computing dependencies
Analyzing 2 sources with 0 local dependencies
Leaving directory '.pytype'

One or more pytype checks failed.
You can view these errors locally by running
  ./build_tools/pytype/check_diff.sh main
Error: Process completed with exit code [12](https://github.com/openxla/iree/actions/runs/6018714036/job/16327379675?pr=14860#step:6:13)3.
```

**Comment by ScottTodd** [comment]
> I keep getting failures on `pytype` check; I can't reproduce it locally. Can I get some eyes from someone or someone can help fix it?
> 
> ```
> Checking .py files in integrations/.*
> ninja: Entering directory `.pytype'
> ninja: error: build.ninja:13: multiple rules generate /home/runner/work/iree/iree/.pytype/pyi/setup.pyi
> 
> Computing dependencies
> Analyzing 2 sources with 0 local dependencies
> Leaving directory '.pytype'
> 
> One or more pytype checks failed.
> You can view these errors locally by running
>   ./build_tools/pytype/check_diff.sh main
> Error: Process completed with exit code [12](https://github.com/openxla/iree/actions/runs/6018714036/job/16327379675?pr=14860#step:6:13)3.
> ```

Any "ninja: error: build.ninja:13: multiple rules generate" style error from pytype can be ignored. It is buggy with our Python/CMake/etc. setup.

**Review by ScottTodd — APPROVED** [review]

**Comment by hanhanW** [comment]
I see, thanks! Can you help land the PR?

**Comment by ScottTodd** [comment]
> I see, thanks! Can you help land the PR?

I think you should be able to merge yourself (the pytype check should be non-blocking)... it at least doesn't warn me that I'm using admin powers when I click the squash and merge button myself :P

**Comment by hanhanW** [comment]
cool, thanks!

**Pull request merged**